app/controllers/spree/api/paypal_controller.rb

```diff
@@ -0,0 +1,202 @@
+module Spree
+  module Api
+    class PaypalController < Spree::Api::BaseController
+      ssl_allowed
+
+      def express
+
+        order = current_order || raise(ActiveRecord::RecordNotFound)
+        items = order.line_items.map(&method(:line_item))
+
+        tax_adjustments = order.all_adjustments.tax.additional
+        shipping_adjustments = order.all_adjustments.shipping
+
+        order.all_adjustments.eligible.each do |adjustment|
+          next if (tax_adjustments + shipping_adjustments).include?(adjustment)
+          items << {
+            :Name => adjustment.label,
+            :Quantity => 1,
+            :Amount => {
+              :currencyID => order.currency,
+              :value => adjustment.amount
+            }
+          }
+        end
+
+        # Because PayPal doesn't accept $0 items at all.
+        # See #10
+        # https://cms.paypal.com/uk/cgi-bin/?cmd=_render-content&content_ID=developer/e_howto_api_ECCustomizing
+        # "It can be a positive or negative value but not zero."
+        items.reject! do |item|
+          item[:Amount][:value].zero?
+        end
+        pp_request = provider.build_set_express_checkout(express_checkout_request_details(order, items))
+
+        # Trying to conform to https://guides.spreecommerce.com/api/summary.html
+        # as much as possible
+        begin
+          pp_response = provider.set_express_checkout(pp_request)
+          if pp_response.success?
+            url = provider.express_checkout_url(pp_response, :useraction => 'commit')
+            
+            response = Spree::Paypal.new
+            response.redirect_url = url
+            respond_with response
```

Comment:
Very strange. In my app it just picks a layout for the json. Doesn't have anything about the view it uses in the log.

It looks like your app is not understanding to render the json even though spree's api ate least purports to do that here

https://github.com/spree/spree/blob/2-4-stable/api/lib/spree/api/controller_setup.rb 


so I guess I need to wrap those `render :json` calls in `respond_to` blocks ala

http://stackoverflow.com/questions/22943892/actioncontrollerunknownformat/22944769#22944769

and/or 

add  `:defaults => { :format => 'json' }`

http://stackoverflow.com/questions/22943892/actioncontrollerunknownformat/28750863#28750863

to those api route definitions